File: pages/inventory_page.py
from playwright.sync_api import Page

class InventoryPage:
    def __init__(self, page: Page):
        self.page = page
        self.inventory_list = ".inventory_list"
        self.add_to_cart_btn = "button[data-test='add-to-cart-sauce-labs-backpack']"
        self.cart_icon = ".shopping_cart_link"
        self.cart_badge = ".shopping_cart_badge"
        # self.product_title = "span[data-test='title']"

    def is_loaded(self):
        return self.page.locator(self.inventory_list).is_visible()
    
    def add_first_item_to_cart(self):
        # self.page.click(self.add_to_cart_btn)
        self.page.locator(self.add_to_cart_btn).first.click()

    def open_cart(self):
        self.page.click(self.cart_icon)

    def items_in_cart(self):
        if self.page.locator(self.cart_badge).is_visible():
            return int(self.page.locator(self.cart_badge).text_content())
        return 0

File: PabloDelgadillo/src/pages/inventoryPage.ts
import { Page } from '@playwright/test';

export class InventoryPage {
  private page: Page;
  private inventoryList = '.inventory_list';
  private addToCartBtn = "button[data-test='add-to-cart-sauce-labs-backpack']";
  private cartIcon = '.shopping_cart_link';
  private cartBadge = '.shopping_cart_badge';

  constructor(page: Page) {
    this.page = page;
  }

  async isLoaded(): Promise<boolean> {
    return await this.page.locator(this.inventoryList).isVisible();
  }

  async addFirstItemToCart() {
    await this.page.locator(this.addToCartBtn).first().click();
  }

  async openCart() {
    await this.page.click(this.cartIcon);
  }

  async itemsInCart(): Promise<number> {
    const badge = this.page.locator(this.cartBadge);
    if (await badge.isVisible()) {
      const text = await badge.textContent();
      return text ? parseInt(text) : 0;
    }
    return 0;
  }
}
